sklearn/preprocessing/data.py

```diff
@@ -1946,3 +1958,202 @@ def transform(self, X):
+    def __init__(self, transformed_features="all", n_jobs=1, copy=True):
+        self.transformed_features = transformed_features
+        self.n_jobs = n_jobs
+        self.copy = copy
+
+    def fit(self, X, y=None):
+        """Estimate lambda for each feature to maximise log-likelihood
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data to fit by apply boxcox transform,
+            to each of the features and learn the lambda.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES)
+        self.n_features_ = X.shape[1]
+        if self.transformed_features is "all":
+            self.transformed_features_ = np.arange(self.n_features_)
+        else:
+            self.transformed_features_ = np.copy(self.transformed_features)
+            if self.transformed_features_.dtype == np.bool:
+                self.transformed_features_ = \
+                    np.where(self.transformed_features_)[0]
+        if any(np.any(X[:, self.transformed_features_] <= 0, axis=0)):
+            raise ValueError("BoxCox transform can only be applied "
+                             "on positive data")
+        out = Parallel(n_jobs=self.n_jobs)(delayed(_boxcox)(X, i,
+                                           lambda_x=None)
+                                           for i in self.transformed_features_)
+        self.lambdas_ = np.array([o[1] for o in out])
+        return self
+
+    def transform(self, X, y=None):
+        """Transform each feature using the lambdas evaluated during fit time
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            The data to apply boxcox transform.
+
+        Returns
+        -------
+        X_tr : array-like, shape (n_samples, n_features)
+            The transformed data.
+        """
+        X = check_array(X, ensure_2d=True, dtype=FLOAT_DTYPES, copy=self.copy)
+        if any(np.any(X[:, self.transformed_features_] <= 0, axis=0)):
+            raise ValueError("BoxCox transform can only be applied "
+                             "on positive data")
+        if X.shape[1] != self.n_features_:
+            raise ValueError("X has a different shape than during fitting.")
+        X_tr = _transform_selected(X, self._transform,
+                                   self.transformed_features_,
+                                   copy=False, order=True)
+        return X_tr
+
+    def _transform(self, X):
+        outputs = Parallel(n_jobs=self.n_jobs)(
+            delayed(_boxcox)(X, i, self.lambdas_[i])
+            for i in range(len(self.transformed_features_)))
+        output = np.concatenate([o[..., np.newaxis] for o in outputs], axis=1)
+        return output
+
+    def inverse_transform(self, X, y=None):
+        """Invere transform each feature using the lambdas evaluated during fit time
```

Comment:
> Good question! Hmm. maybe that diff isn't working quite right.

Nothing to do with the diff, flake8 is not complaining about this line, quickly looking at it it looks like flake8 is measuring the length of the line with the spaces stripped at the beginnning.